backend/src/main/java/gov/cdc/usds/simplereport/db/model/TestEvent.java

```diff
@@ -46,37 +46,35 @@
 
 	public TestEvent() {}
 
-	public TestEvent(TestResult result, DeviceType deviceType, Person patient, Facility facility, TestOrder order) {
-		super(patient, facility, deviceType, result);
-		// store a link, and *also* store the object as JSON
-		this.patientData = getPatient();
-		this.providerData = getFacility().getOrderingProvider();
-		this.order = order;
-		super.setDateTestedBackdate(order.getDateTestedBackdate());
-		PatientAnswers answers = order.getAskOnEntrySurvey();
-		if (answers != null) {
-			this.surveyData = order.getAskOnEntrySurvey().getSurvey();
-		} else {
-			// this can happen during unit tests, but never in prod.
-			LOG.error("Order {} missing PatientAnswers", order.getInternalId());
-		}
-	}
-
-	public TestEvent(TestOrder order) {
-		this(order.getResult(), order.getDeviceType(), order.getPatient(), order.getFacility(), order);
+    public TestEvent(TestResult result, DeviceSpecimen deviceType, Person patient, Facility facility, TestOrder order) {
+        super(patient, facility, deviceType, result);
+        // store a link, and *also* store the object as JSON
+        this.patientData = getPatient();
+        this.providerData = getFacility().getOrderingProvider();
+        this.order = order;
+        super.setDateTestedBackdate(order.getDateTestedBackdate());
+        PatientAnswers answers = order.getAskOnEntrySurvey();
+        if (answers != null) {
+            this.surveyData = order.getAskOnEntrySurvey().getSurvey();
+        } else {
+            // this can happen during unit tests, but never in prod.
+            LOG.error("Order {} missing PatientAnswers", order.getInternalId());
+        }
+    }
+
+    public TestEvent(TestOrder order) {
+        this(order.getResult(), order.getDeviceSpecimen(), order.getPatient(), order.getFacility(), order);
 	}
 
 	// Constructor for creating corrections. Copy the original event
 	public TestEvent(TestEvent event, TestCorrectionStatus correctionStatus, String reasonForCorrection) {
 		super(event, correctionStatus, reasonForCorrection);
 
-		this.order = event.getTestOrder();
 		this.patientData = event.getPatientData();
 		this.providerData = event.getProviderData();
 		this.order = event.getTestOrder();
-		// this.patient_answers_data =
 		this.surveyData = event.getSurveyData();
-		super.setDateTestedBackdate(order.getDateTestedBackdate());
+        setDateTestedBackdate(order.getDateTestedBackdate());
```

Comment:
The `this` is not needed here but lots of people like to use it in constructors (and it is very much needed when you shadow your field names with parameter names, which lots of code generators love to do). The `super` however is IMO harmful: right now it is going to hit the correct method, since there is no override of that method in this class, and if we were to add an override, I would think we'd expect it to be called here, rather than being bypassed.